contracts/contracts/Adjudicator.sol

```diff
@@ -6,14 +6,15 @@ import "./lib/ReEncryptionValidator.sol";
 import "./lib/SignatureVerifier.sol";
 import "@openzeppelin/contracts/utils/math/Math.sol";
 import "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import "./TACoApplication.sol";
 
 
 /**
 * @title Adjudicator
 * @notice Supervises operators' behavior and punishes when something's wrong.
 * @dev |v3.1.1|
 */
-abstract contract Adjudicator {
+contract Adjudicator {
```

Comment:
It seems `Adjudicator` is not upgradeable anymore, right? In that case, the `uint256[50] private reservedSlots;` in L46 should be removed.

Reply:
Honestly, `Adjudicator` needs reincarnation but I guess it's not time. So I'll remove those fields but ideally make it upgradeable

Reply:
Adjudicator has some state that ideally to preserve. For example, history to not slash twice for the same misbehavior. That's main reason to make it upreadable with the current design

Reply:
Tracking issue here: https://github.com/nucypher/nucypher-contracts/issues/100